rbac/management/tenant_service/v2.py

```diff
@@ -0,0 +1,392 @@
+                "parent",
+            ),
+            create_relationship(
+                ("rbac", "workspace"), str(root_workspace.uuid), ("rbac", "tenant"), tenant_id, "parent"
+            ),
+        ]
+
+        # Include platform for tenant
+        relationships.append(
+            create_relationship(("rbac", "tenant"), tenant_id, ("rbac", "platform"), settings.ENV_NAME, "platform")
+        )
+
+        mapping = TenantMapping.objects.create(tenant=tenant)
+        relationships.extend(self._bootstrap_default_access(tenant, mapping, default_workspace))
+
+        self._replicator.replicate(
+            ReplicationEvent(
+                event_type=ReplicationEventType.BOOTSTRAP_TENANT,
+                info={"org_id": tenant.org_id, "default_workspace_uuid": str(default_workspace.uuid)},
+                partition_key="rbactodo",
+                add=relationships,
+            )
+        )
+
+        return BootstrappedTenant(tenant, mapping)
+
+    def _bootstrap_default_access(
+        self, tenant: Tenant, mapping: TenantMapping, default_workspace: Workspace
+    ) -> List[Relationship]:
+        """
+        Bootstrap default access for a tenant's users and admins.
+
+        Creates role bindings between the tenant's default workspace, default groups, and system policies.
+        """
+        platform_default_role = self._get_platform_default_policy_uuid()
+        admin_default_role = self._get_admin_default_policy_uuid()
+
+        if platform_default_role is None:
+            logger.warning("No platform default role found for public tenant. Default access will not be set up.")
+
+        if admin_default_role is None:
+            logger.warning("No admin default role found for public tenant. Default access will not be set up.")
+
+        default_workspace_uuid = str(default_workspace.uuid)
+        default_user_role_binding_uuid = str(mapping.default_user_role_binding_uuid)
+        default_admin_role_binding_uuid = str(mapping.default_admin_role_binding_uuid)
+
+        tuples_to_add: List[Relationship] = []
+
+        # Add default role binding IFF there is no custom default access for the tenant
+
+        # NOTE: This logic is prone to write skew: the platform group for the tenant may be created concurrently.
+        # Care must be taken to prevent this.
+        # Currently, when this default group is created (in group/definer.py) we:
+        # 1. Check for the existence of a tenant mapping. If exists, use that.
+        #    No race because if it already exists, this process must've already happened.
+        # 2. If tenant mapping does not exist, create it via this same bootstrap process.
+        #    Due to unique constraint, if this happens concurrently from another input (e.g. user import),
+        #    one will rollback, serializing the group creation with user import on next retry.
+        if platform_default_role and not Group.objects.filter(platform_default=True, tenant=tenant).exists():
```

Comment:
Isn't `platform_default_role` always set ? (same for admin_default_role )

Reply:
It's not always set in tests unfortunately. So rather than update a bunch of tests, I added this warning. I can see how many tests it is maybe it's not too bad to update them.